src/devices/Ft232H/Ft232HDevice.cs

```diff
@@ -0,0 +1,857 @@
+                    toSend[idx++] = _gpioLowDir;
+                }
+            }
+            else
+            {
+                _gpioHighDir |= (byte)(1 << (chipSelect - 8));
+                if (value == PinValue.High)
+                {
+                    _gpioHighData |= (byte)(1 << (chipSelect - 8));
+                }
+                else
+                {
+                    byte mask = 0xFF;
+                    mask &= (byte)(~(1 << (chipSelect - 8)));
+                    _gpioHighData &= mask;
+                }
+
+                for (int i = 0; i < NumberCycles; i++)
+                {
+                    toSend[idx++] = (byte)FtOpcode.SetDataBitsHighByte;
+                    toSend[idx++] = _gpioHighData;
+                    toSend[idx++] = _gpioHighDir;
+                }
+            }
+
+            Write(toSend);
+        }
+
+        #endregion
+
+        #region Read Write
+
+        internal void Write(ReadOnlySpan<byte> buffer)
+        {
+            uint numBytesWritten = 0;
+            var ftStatus = FtFunction.FT_Write(_ftHandle, in MemoryMarshal.GetReference(buffer), (ushort)buffer.Length, ref numBytesWritten);
+            if ((ftStatus != FtStatus.Ok) || (buffer.Length != numBytesWritten))
+            {
+                throw new IOException($"Can't write to the device");
+            }
+        }
+
+        internal int Read(Span<byte> buffer)
+        {
+            CancellationToken token = new CancellationTokenSource(1000).Token;
+            int totalBytesRead = 0;
+            uint bytesToRead = 0;
+            uint numBytesRead = 0;
+            FtStatus ftStatus;
+            while ((totalBytesRead < buffer.Length) && (!token.IsCancellationRequested))
+            {
+                bytesToRead = GetAvailableBytes();
+                if (bytesToRead > 0)
+                {
+                    ftStatus = FtFunction.FT_Read(_ftHandle, in buffer[totalBytesRead], bytesToRead, ref numBytesRead);
+                    if ((ftStatus != FtStatus.Ok) && (bytesToRead != numBytesRead))
+                    {
+                        throw new IOException("Can't read device");
+                    }
+
+                    totalBytesRead += (int)bytesToRead;
+                }
+            }
+
+            return totalBytesRead;
+        }
+
+        private bool FlushBuffer()
+        {
+            var availableBytes = GetAvailableBytes();
+
+            if (availableBytes > 0)
+            {
+                byte[] toRead = new byte[availableBytes];
+                uint bytesRead = 0;
+                var ftStatus = FtFunction.FT_Read(_ftHandle, in toRead[0], availableBytes, ref bytesRead);
+                return ftStatus == FtStatus.Ok;
+            }
+
+            return true;
+        }
+
+        private uint GetAvailableBytes()
+        {
+            uint availableBytes = 0;
+            var ftStatus = FtFunction.FT_GetQueueStatus(_ftHandle, ref availableBytes);
```

Comment:
the name suggest the queue could mean either read or write queue but you always use it in combination with reading, I'd suggest changing the name of this method and explain that i.e.: write queue will always be empty because we wait on write so this actually always means read queue

Reply:
Those are the name of the FTDI API. I can adjust them but for reference to the docs, I would prefer to keep them like they are to make it easier later to debug or implement something new.
And you are correct about what it does, it only checks how many bytes are available to read. I'll adjust the docs of the function.